packages/credential-provider-ini/src/index.ts

```diff
@@ -115,16 +117,33 @@ const isWebIdentityProfile = (arg: any): arg is WebIdentityProfile =>
   typeof arg.web_identity_token_file === "string" &&
   typeof arg.role_arn === "string" &&
   ["undefined", "string"].indexOf(typeof arg.role_session_name) > -1;
-interface AssumeRoleProfile extends Profile {
+
+interface AssumeRoleWithSourceProfile extends Profile {
   role_arn: string;
   source_profile: string;
 }
 
-const isAssumeRoleWithSourceProfile = (arg: any): arg is AssumeRoleProfile =>
+interface AssumeRoleWithProviderProfile extends Profile {
+  role_arn: string;
+  credential_source: string;
+}
+
+const isAssumeRoleWithSourceProfile = (arg: any): arg is AssumeRoleWithSourceProfile =>
   Boolean(arg) &&
   typeof arg === "object" &&
   typeof arg.role_arn === "string" &&
   typeof arg.source_profile === "string" &&
+  typeof arg.credential_source === "undefined" &&
+  ["undefined", "string"].indexOf(typeof arg.role_session_name) > -1 &&
+  ["undefined", "string"].indexOf(typeof arg.external_id) > -1 &&
+  ["undefined", "string"].indexOf(typeof arg.mfa_serial) > -1;
+
+const isAssumeRoleWithProviderProfile = (arg: any): arg is AssumeRoleWithProviderProfile =>
+  Boolean(arg) &&
+  typeof arg === "object" &&
+  typeof arg.role_arn === "string" &&
+  typeof arg.credential_source === "string" &&
+  typeof arg.source_profile === "undefined" &&
   ["undefined", "string"].indexOf(typeof arg.role_session_name) > -1 &&
   ["undefined", "string"].indexOf(typeof arg.external_id) > -1 &&
   ["undefined", "string"].indexOf(typeof arg.mfa_serial) > -1;
```

Comment:
Move common code to a different function?

```suggestion
const isAssumeRoleProfile = (arg: any) =>
  Boolean(arg) &&
  typeof arg === "object" &&
  typeof arg.role_arn === "string" &&
  ["undefined", "string"].indexOf(typeof arg.role_session_name) > -1 &&
  ["undefined", "string"].indexOf(typeof arg.external_id) > -1 &&
  ["undefined", "string"].indexOf(typeof arg.mfa_serial) > -1;

const isAssumeRoleWithSourceProfile = (arg: any): arg is AssumeRoleWithSourceProfile =>
  isAssumeRoleProfile(arg) &&
  typeof arg.source_profile === "string" &&
  typeof arg.credential_source === "undefined";

const isAssumeRoleWithProviderProfile = (arg: any): arg is AssumeRoleWithProviderProfile =>
  isAssumeRoleProfile(arg) &&
  typeof arg.credential_source === "string" &&
  typeof arg.source_profile === "undefined";
```